src/MacroTools/Extensions/RectangleExtensions.cs

```diff
@@ -22,53 +22,62 @@ public static void AddSound(this Rectangle region, sound soundHandle)
       SetSoundPosition(soundHandle, GetRectCenterX(region.Rect), GetRectCenterY(region.Rect), 0);
       RegisterStackedSound(soundHandle, true, width, height);
     }
-    
+
     /// <summary>
-    /// Prepares all Neutral Passive owned units within the specified <paramref name="rectangle"/> according to
+    /// Prepares units owned by <paramref name="owningPlayer"/> within the specified <paramref name="rectangle"/> according to
     /// the provided <see cref="RescuePreparationMode"/>.
     /// </summary>
     /// <param name="rectangle">The rectangle in which to prepare units.</param>
     /// <param name="rescuePreparationMode">Determines how units are prepared.</param>
-    /// <returns>All Neutral Passive units in the specified rectangle.</returns>
-    public static List<unit> PrepareUnitsForRescue(this Rectangle rectangle, RescuePreparationMode rescuePreparationMode)
+    /// <param name="owningPlayer">The player owning the units inside <paramref name="rectangle"/></param>
+    /// <param name="filter">A function that is applied to each unit found in <paramref name="rectangle"/>
+    /// <para/>
+    /// The unit gets rescued if <paramref name="filter"/> returns true, else it does not get rescued.
+    /// </param>
+    /// <returns>Returns all <paramref name="owningPlayer"/> units in the specified rectangle. Default is neutral passive.</returns>
+    public static List<unit> PrepareUnitsForRescue(this Rectangle rectangle, RescuePreparationMode rescuePreparationMode, player? owningPlayer = null, Func<unit, bool>? filter = null)
```

Comment:
Sorry, this might be obvious but - could you clarify what problem the new `owningPlayer` parameter is meant to solve? Do we currently have a Quest which needs this?

Reply:
Yes, `QuestHellFireCitadel` requires it because all units start hostile.
We had a similar case when I implemented the "Scepter of The Queen" Quests, but that got fixed in a different way but I can't remember how.

Reply:
Ok, makes sense. `QuestHellfireCitadel` starting hostile is actually a bug - so I think the cleaner move here is to just fix that bug and remove the player parameter from this one. We should be able to guarantee that *all* rescuable units start neutral passive.